jolt-core/src/poly/commitment/dory/wrappers.rs

```diff
@@ -0,0 +1,331 @@
+            }
+            MultilinearPolynomial::OneHot(poly) => {
+                let mut result = vec![Fr::zero(); num_cols];
+                poly.vector_matrix_product(&wrapped_left_side, Fr::one(), &mut result);
+                result.into_iter().map(|v| jolt_to_ark(&v)).collect()
+            }
+            MultilinearPolynomial::RLC(poly) => poly
+                .vector_matrix_product(&wrapped_left_side)
+                .into_iter()
+                .map(|v| jolt_to_ark(&v))
+                .collect(),
+        }
+    }
+}
+
+fn commit_tier_1<E>(
+    poly: &MultilinearPolynomial<Fr>,
+    g1_generators: &[E::G1],
+    row_len: usize,
+) -> Result<Vec<E::G1>, DoryError>
+where
+    E: PairingCurve,
+    E::G1: DoryGroup<Scalar = ArkFr>,
+{
+    // SAFETY: E::G1 and ArkG1 have the same memory layout when E = BN254.
+    let g1_slice = unsafe {
+        std::slice::from_raw_parts(g1_generators.as_ptr() as *const ArkG1, g1_generators.len())
+    };
+
+    // All polynomial types should use row_len bases (number of columns).
+    // The globals are sized to be >= what any polynomial needs.
+    let bases: Vec<G1Affine> = g1_slice
+        .iter()
+        .take(row_len)
+        .map(|g| g.0.into_affine())
+        .collect();
+
+    macro_rules! compute_msm {
+        ($coeffs:expr, $msm_method:ident) => {
+            $coeffs
+                .par_chunks(row_len)
+                .map(|row| ArkG1(VariableBaseMSM::$msm_method(&bases[..row.len()], row).unwrap()))
+                .collect()
+        };
+    }
+
+    let result: Vec<ArkG1> = match poly {
+        MultilinearPolynomial::LargeScalars(poly) => {
+            compute_msm!(&poly.Z, msm_field_elements)
+        }
+        MultilinearPolynomial::U8Scalars(poly) => compute_msm!(&poly.coeffs, msm_u8),
+        MultilinearPolynomial::U16Scalars(poly) => compute_msm!(&poly.coeffs, msm_u16),
+        MultilinearPolynomial::U32Scalars(poly) => compute_msm!(&poly.coeffs, msm_u32),
+        MultilinearPolynomial::U64Scalars(poly) => compute_msm!(&poly.coeffs, msm_u64),
+        MultilinearPolynomial::I64Scalars(poly) => compute_msm!(&poly.coeffs, msm_i64),
+        MultilinearPolynomial::I128Scalars(poly) => compute_msm!(&poly.coeffs, msm_i128),
+        MultilinearPolynomial::U128Scalars(poly) => compute_msm!(&poly.coeffs, msm_u128),
+        MultilinearPolynomial::S128Scalars(poly) => compute_msm!(&poly.coeffs, msm_s128),
+        MultilinearPolynomial::BoolScalars(poly) => poly
+            .coeffs
+            .par_chunks(row_len)
+            .map(|row| {
+                let field_row: Vec<Fr> = row
+                    .iter()
+                    .map(|&b| if b { Fr::from(1u64) } else { Fr::from(0u64) })
+                    .collect();
+                ArkG1(
+                    VariableBaseMSM::msm_field_elements(&bases[..field_row.len()], &field_row)
+                        .unwrap(),
+                )
```

Comment:
nit: I know we aren't using this, but using `msm_u8` (or computing the boolean MSM inline here) would be better